docs: add note about notifications on real devices (#3592)
invertase/react-native-firebase: docs/messaging/notifications.md

The call to `getInitialNotification` should happen within a React lifecycle method after mounting (e.g. `componentDidMount` or `useEffect`).
If it's called too soon (e.g. within a class constructor or global scope), the notification data may not be available.

> **Quick Tip:** On `Android` you can test this interactions on the emulator but on `iOS` you need to use a real device because the iOS simulator doesn't receive push notifications.

Reviewer: ```suggestion
**Quick Tip:** On `Android` you can test receiving remote notifications on the emulator but on `iOS` you will need to use a real device as the iOS simulator does not support receiving remote notifications.
```
